app/services/statements/query.rb

```diff
@@ -0,0 +1,61 @@
+module Statements
+  class Query
+    include Queries::ConditionFormats
+    include API::Concerns::FilterIgnorable
+
+    attr_reader :scope
+
+    def initialize(lead_provider: :ignore, registration_period_start_years: :ignore, updated_since: :ignore, state: :ignore, output_fee: true)
+      @scope = Statement.distinct.includes(:lead_provider, :registration_period)
+
+      where_lead_provider_is(lead_provider)
+      where_registration_period_start_year_in(registration_period_start_years)
+      where_updated_since(updated_since)
+      where_state_is(state)
+      where_output_fee_is(output_fee)
+    end
+
+    def statements
+      scope.order(payment_date: :asc)
+    end
+
+    def statement(id: nil, api_id: nil)
+      return scope.find_by!(api_id:) if api_id.present?
+      return scope.find(id) if id.present?
+
+      fail(ArgumentError, "id or api_id needed")
+    end
```

Comment:
this might require more discussion, not a huge fan of the multiple id options.
Should we break it into two methods instead?
`def statement_by_api_id`

Reply:
Yeah agree.. tks @mooktakim I've made the change.. I can't see the need of looking for a statement other than by `api_id` in ECF2. Maybe we had this need in ECF1 as lead provider would search by either id or ecf_id when we flipped ids..